components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -0,0 +1,628 @@
+        }
+
+        ApplicationManagementService applicationMgtService = OAuth2ServiceComponentHolder.getApplicationMgtService();
+        String spName = applicationMgtService
+                .getServiceProviderNameByClientId(requestMsgCtx.getAuthorizationReqDTO().getConsumerKey(),
+                        INBOUND_AUTH2_TYPE, spTenantDomain);
+        ServiceProvider serviceProvider = applicationMgtService.getApplicationExcludingFileBasedSPs(spName,
+                spTenantDomain);
+        if (serviceProvider == null) {
+            return mappedAppClaims;
+        }
+        ClaimMapping[] requestedLocalClaimMap = serviceProvider.getClaimConfig().getClaimMappings();
+        if (requestedLocalClaimMap == null || !(requestedLocalClaimMap.length > 0)) {
+            return new HashMap<>();
+        }
+
+        AuthenticatedUser user = requestMsgCtx.getAuthorizationReqDTO().getUser();
+        String userTenantDomain = user.getTenantDomain();
+        String username = user.toString();
+        UserRealm realm = IdentityTenantUtil.getRealm(userTenantDomain, username);
+        if (realm == null) {
+            log.warn("Invalid tenant domain provided. Empty claim returned back for tenant " + userTenantDomain
+                    + " and user " + user);
+            return new HashMap<>();
+        }
+
+        List<String> claimURIList = new ArrayList<>();
+        for (ClaimMapping mapping : requestedLocalClaimMap) {
+            if (mapping.isRequested()) {
+                claimURIList.add(mapping.getLocalClaim().getClaimUri());
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Requested number of local claims: " + claimURIList.size());
+        }
+
+        Map<String, String> spToLocalClaimMappings = ClaimMetadataHandler.getInstance().
+                getMappingsMapFromOtherDialectToCarbon(OIDC_DIALECT, null, spTenantDomain, false);
+
+        Map<String, String> userClaims = null;
+        try {
+            userClaims = realm.getUserStoreManager().getUserClaimValues(UserCoreUtil.addDomainToName(
+                    user.getUserName(), user.getUserStoreDomain()), claimURIList.toArray(
+                    new String[claimURIList.size()]), null);
+        } catch (UserStoreException e) {
+            if (e.getMessage().contains("UserNotFound")) {
+                if (log.isDebugEnabled()) {
+                    log.debug("User " + user + " not found in user store");
+                }
+            } else {
+                throw e;
+            }
+        }
+
+        if (isEmpty(userClaims)) {
+            // User claims can be empty if user does not exist in user stores. Probably a federated user.
+            if (log.isDebugEnabled()) {
+                log.debug("No claims found for " + username + " from user store.");
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Number of user claims retrieved from user store: " + userClaims.size());
```

Comment:
Possible NullPointer. Better to  null chcek

Reply:
else {} is only executed when userClaim map is not null